app/controllers/users/reset_passwords_controller.rb

```diff
@@ -49,7 +53,7 @@ def update
         success: result.success?,
         failure_reason: irs_attempts_api_tracker.parse_failure_reason(result),
       )
-
+      session.delete(:reset_password_token) if session[:reset_password_token]
```

Comment:
Should we only delete it if it was successful?

Reply:
no, if its not successful we redirect the user back to the forgot password page. not the same page, so the user would have to reenter in their email and reconfirm anyways, this would reduce the chance of confusion if we delete it.